rfd/0009-viewer-app.md

```diff
@@ -0,0 +1,164 @@
+
+#### 1. Cross-Site Scripting (XSS)
+* XSS is a serious client-side vulnerability. A perpetrator is able to add some malicious code to to our program that is interpreted as valid and is executed as a part of the application. This compromises the functionality of the app and the user data. Luckily, React outputs elements and data inside them using auto escaping. We can further secure our app by using a library such as [DOMPurify](https://github.com/cure53/DOMPurify). 
+
+    * Validate all data that flows into app from the server or a third-party API. This cushions the app against an XSS attack, and at times, you may be able to prevent it, as well.
+
+    * Don't mutate DOM directly. If you need to render different content, use innerText instead of innerHTML. Be extremely cautious when using escape hatches like findDOMNode or createRef in React.
+
+    * Always try to render data through JSX and let React handle the security concerns for you.
+
+    * Avoid writing your own sanitization techniques. It's a separate subject on its own that requires some expertise.
+
+    * Use good libraries for sanitizing your data. There are a number of them, but you must compare the pros and cons of each specific to your use case before going forward with one.
+
+#### 2. Broken Authentication
+* Another common issue in React.js applications is inadequate or poor authorization. This can result in attackers hacking user credentials and carrying out brute force attacks. There are various risks associated with broken authorization, like session IDs being exposed in URLs, easy and predictable login details being discovered by attackers, the unencrypted transmission of credentials, persisting valid sessions after logout, and other session-related factors.
+
+We can implement safe session management. A lot of times developers store session-id in front-end URLs, making it clearly visible to users as well as attackers. Instead, you could store the session-id inside browser storage and use it via a custom React hook in whichever component or page of your app where you need it. We can facilitate this by creating a `useSession` hook. 
+
+```javascript
+import { useEffect, useState } from "react"
+
+export default function useSession({session_id}){
+
+    const [sessionId,setSessionId]=useState()
+
+    useEffect(()=>{
+        if(session_id) setSessionId(session_id);
+        else{
+            if(localStorage.session_id) setSessionId(localStorage.session_id)
+        }
+    },[])
+
+    useEffect(()=>{
+        if(!localStorage.session_id) localStorage.setItem('session_id',sessionId)
+    },[sessionId])
+
+    return{
+        sessionId
+    }
+}
+```
+
+So now whenever you need to use session ID on a particular page, you can grab it from this hook. Thus, you won't need to pass it as query parameters in your React app's routes. 
```

Comment:
Thanks for adding that. What about on the server side, how does the session token work and how do you implement logouts?